### DISCUSSION TITLE
Dexcom G6: signal missed very often with new phone (MIUI 13.0.3)

### DISCUSSION THREAD

**Discussion by dagord** [NightscoutFoundation/xDrip#2464] [Q&A]
Hello
I'm using Dexcom G6 with xDrip+ for the last months without big issues on my **Xiaomi Redmi Note 7 with MIUI 12.5.1**
I few days ago I switched to a **Xiaomi Redmi Note 10 with MIUI 13.0.3** and since then I'm experiencing a lot of short disconnections.
The scan interval should be 5 minutes but I get very very often a "signal missed" message, even with the phone at my side. The connection is usually restored without any intervention within 15 minutes from the previous scan; I'm not sure if the collector is restarted by the app or the connection gets back by itself.

I'm using the same xDrip+ version (2122080314); I replicated the settings using the qrcode (I double checked them); I disabled the MIUI battery management service for xDrip+ as well.

I specify that in the meanwhile the old trasmitter expired so maybe the new one is defective... but I'm suspecting some system settings.

Attaching some screenshots and logs

![Screenshot_2022-11-15-10-26-55-084_com eveningoutpost dexdrip](https://user-images.githubusercontent.com/21258057/201889523-bbdf70f9-4114-4b92-90f1-d19bda31cd84.jpg)
![Screenshot_2022-11-15-10-43-49-919_com miui gallery](https://user-images.githubusercontent.com/21258057/201889543-bed094a8-9a3f-4f1e-bb70-55db510aa423.jpg)
![Screenshot_2022-11-15-10-19-54-541_com eveningoutpost dexdrip](https://user-images.githubusercontent.com/21258057/201889557-bc54172f-3940-4d20-9482-e8bd23ee4c69.jpg)

**Comment by Navid200** [comment]
Can you find settings on your phone described in the following for Xiaomi?
https://dontkillmyapp.com/

**Reply by dagord** [reply]
Hello.
I read that website but it refers to MIUI up to version 12.*, while I'm running MIUI 13.0.3
I tried to replicate all settings but maybe something is different

**Reply by dagord** [reply]
Tried with the last nightly but the issue persists

**Reply by Navid200** [reply]
There has been no recent changes to the G6 collector.

Xiaomi is a known problem brand.  You may not find your model on that site.  Use it the best way you can and find settings you can to stop the phone from disabling xDrip in the background.

**Reply by katpetrova** [reply]
Hi. What is working for me, different Xiaomi but again with Miui 13: used the directions to become 'developer'. Press on Miui few times until it says tou are developer. Then found 'xdrip' and it wasn't active. Activated it and for now seems issue solved (one day is working without interruption)

**Reply by dagord** [reply]
Sorry, I don't understand your directions.
I'm already in developer mode: where should I enable xDrip?

**Reply by katpetrova** [reply]
On general settings, type 'xdrip' and check if it is enabled. On my phone wasn't

**Reply by dagord** [reply]
Ok, I did as you suggested (anyway, I don't know what it is supposed to do)
At first it seemed to fix the issue but after a couple of hours it was back again.

There are other users having similar problems with Dexcom tramsmitter using Android 12 so I suppose there is some issue related to bluetooth power management, but I don't find a way to disable it as it was on previous android versions!

**Reply by Miquest** [reply]
Have you tried the Broadcast to xDrip Plus from the Dexcom BYODA Application?
It works on my Phone ( Oppo Find X5 Pro, ColorOS 12.1).
It seems like the Dexcom app doesn't get killed as quickly by the system as xDrip, so I think this is a good workaround that seems to work at least on Oppo devices.

**Reply by Miquest** [reply]
You can access the BYODA from this link:
https://docs.google.com/forms/d/e/1FAIpQLScD76G0Y-BlL4tZljaFkjlwuqhT83QlFM5v6ZEfO7gCU98iJQ/viewform?fbzx=2196386787609383750&fbclid=IwAR2aL8Cps1s6W8apUVK-gOqgGpA-McMPJj9Y8emf_P0-_gAsmJs6QwAY-o0

Could also be helpful;) :
https://androidaps.readthedocs.io/de/latest/Hardware/DexcomG6.html

**Reply by rudysroost** [reply]
The BYODA workaround mentioned by @thorinEichenschild007 worked for me for about a week, and then that stopped getting the signal too. I tried a reinstall, and it never even finds the transmitter. Dexcom sent me a new receiver because my original had stopped working. There are no issues with the receiver. My current collector is a cheap Android device I found on Amazon running Android 8.0 with my Pixel 7 (Android 13) in Dexcom follow mode.

**Reply by dagord** [reply]
Yes, I've used BYODA in the past, but I'd rather avoid complex setup if possible :)
Actually, xDrip is correctly open and running but the bluetooth connection gets continuously "freezed" for 15mins.
In the meanwhile I found another thread describing a very similar issue: https://github.com/NightscoutFoundation/xDrip/discussions/1951

So for the moment I'm testing "Blu Car Tethering" an application that somehow "refreshes" the bluetooth connection

**Reply by dagord** [reply]
> The BYODA workaround mentioned by @thorinEichenschild007 worked for me for about a week, and then that stopped getting the signal too. I tried a reinstall, and it never even finds the transmitter. Dexcom sent me a new receiver because my original had stopped working. There are no issues with the receiver. My current collector is a cheap Android device I found on Amazon running Android 8.0 with my Pixel 7 (Android 13) in Dexcom follow mode.

So do you think it's an issue with bluetooth as well?

**Reply by rudysroost** [reply]
> So do you think it's an issue with Bluetooth as well?

Yes, I do. I'm baffled by how everything works in the beginning but then the issue gets worse and worse then does not work at all. This makes no sense to me. When the issue began, I would get one reading after rebooting my phone and then no more readings until the next time I rebooted. Eventually, rebooting my phone solved nothing.

**Reply by Miquest** [reply]
> > So do you think it's an issue with Bluetooth as well?
> 
> Yes, I do. I'm baffled by how everything works in the beginning but then the issue gets worse and worse then does not work at all. This makes no sense to me. When the issue began, I would get one reading after rebooting my phone and then no more readings until the next time I rebooted. Eventually, rebooting my phone solved nothing.

Did you use companion mode or set the data source to 640G/Eversense as described? Or activated the "10 Minute WarmUp" option? Maybe that crashed the transmitter.

**Reply by Miquest** [reply]
I don't know if my method works on other phones, but obviously not.

**Comment by Miquest** [comment]
![Screenshot_2022-11-21-18-15-04-32_2dd2c1bcb8f659cd696bcba2072f751b](https://user-images.githubusercontent.com/91971937/203118748-714da609-f5f4-4c8e-9391-49756ce809a5.jpg)
![Screenshot_2022-11-21-18-14-51-21_a4159468672dd071435c46bf7c20daac](https://user-images.githubusercontent.com/91971937/203118753-9e390f50-43d3-4750-a5c4-e4ccbf25141a.jpg)

I could send you the .apk file that worked for me, so that you can try it.

**Reply by rudysroost** [reply]
The Dexcom app no longer connects to the transmitter, it was a good workaround until it slowly failed in the same way xDrip did.

**Reply by Miquest** [reply]
That's strange. Which smartphone do or did you use for this?

Sorry, I just remembered, the Pixel 7 user 🙃

**Reply by Miquest** [reply]
Maybe it's another problem as in my case, I had too long wake Up intervals on my OnePlus before and on my Find X5 Pro which I have since 1 Month. 
Wakeup Times from the OB1 collection service around 5 minutes

**Reply by Navid200** [reply]
> The Dexcom app no longer connects to the transmitter, ,,,.

which confirms the problem is with the phone and not the app.
If the phone does not allow an app to run in the background, there is not much you can do inside the app to remedy the problem.

**Reply by rudysroost** [reply]
> That's strange. Which smartphone do or did you use for this?
> 
> Sorry, I just remembered, the Pixel 7 user 🙃

Yes, Pixel 7, but also with my Pixel 5 after upgrading from Android 12 to Android 13; I was hoping the new phone would resolve the issue, but it seems to me to be an issue with OS version 13.

**Reply by dagord** [reply]
I've other bluetooth devices (Miband 6, speakers, car hi-fi system) and they just work fine with the new phone, so I think the problem is between Dexcom transmitter and Android 12; maybe it's not fully compliant with the new bluetooth stack? Just guessing.
By the way, I've read some threads on Reddit and other users suggest to use Dexcom BYODA app along with Xdrip as well, and I decided to give it a try: it seems to work, in the sense that I don't see any signal drop/reconnection issue, but I'd rather avoid to keep two applications open and running if there is a simpler solution.

At the moment I'm testing "Blu Car Tethering", which purpose is to keep the bt connection alive: it improves the stability, so I can get a reading every 5/10 minutes (even if it should be exactly every 5 minutes) without getting the "missing signal" warning... And if I'm away for more than 30 minutes the connection is restored as soon as I came back, while previously I was asked to re-pair the transmitter.
I need to do long-term tests... but for the moment this helps even if it doesn't solve the issue.
